src/VisualStudio/IntegrationTest/IntegrationTests/AbstractUpdateProjectTest.cs

```diff
@@ -0,0 +1,53 @@
+﻿// Copyright (c) Microsoft.  All Rights Reserved.  Licensed under the Apache License, Version 2.0.  See License.txt in the project root for license information.
+
+using System.IO;
+using System.Linq;
+using System.Xml.Linq;
+using Microsoft.CodeAnalysis.Shared.TestHooks;
+using Microsoft.VisualStudio.IntegrationTest.Utilities;
+using Xunit;
+using ProjectUtils = Microsoft.VisualStudio.IntegrationTest.Utilities.Common.ProjectUtils;
+
+namespace Roslyn.VisualStudio.IntegrationTests
+{
+    public abstract class AbstractUpdateProjectTest : AbstractIntegrationTest
+    {
+        protected AbstractUpdateProjectTest(VisualStudioInstanceFactory instanceFactory)
+            : base(instanceFactory)
+        {
+        }
+
+        protected XElement GetProjectFileElement(ProjectUtils.Project project)
+        {
+            // Save the project file.
+            VisualStudio.SolutionExplorer.SaveAll();
+            VisualStudio.Workspace.WaitForAsyncOperations(FeatureAttribute.Workspace);
```

Comment:
@sharwell do you have insight on the best ways for integration tests to be written to be as stable and non-flakey as possible?  Thanks!